Default selectId function for EntityAdapter

PhillipZada opened the pull request:
See #378 

**What is the current behavior?**

I have several entities, and I use always id property for the primary key. So, to create the EntityAdater I must repeat the code like:

```
export const myEntityAdapter: EntityAdapter<MyEntity> = createEntityAdapter<MyEntity>({
  selectId: (myEntity: MyEntity) => myEntity.id
});
```

**Expected behavior:**

This could be greate if createEntityAdapter uses a default selectId method which will take id property of object as a primary key. So I just need call this function

```
export const myEntityAdapter: EntityAdapter<MyEntity> = createEntityAdapter<MyEntity>();
```

brandonroberts commented on modules/entity/src/create_adapter.ts:
```diff
@@ -10,10 +10,12 @@ import { createSelectorsFactory } from './state_selectors';
 import { createUnsortedStateAdapter } from './unsorted_state_adapter';
 
-export function createEntityAdapter<T>(options: {
+export function createEntityAdapter<T>(options?: {
```
Instead of making `options` optional, make `selectId` optional and provide a default below

brandonroberts commented on modules/entity/src/create_adapter.ts:
```diff
@@ -10,10 +10,12 @@ import { createSelectorsFactory } from './state_selectors';
+  options = options || { selectId: (instance: any) => instance.id };
+
   const { selectId, sortComparer }: EntityDefinition<T> = {
     sortComparer: false,
```
Added a default `selectId` function here under the `sortComparer`

PhillipZada replied:
Done

brandonroberts commented on modules/store/src/store_module.ts:
```diff
@@ -149,12 +149,16 @@ export class StoreModule {
   ): ModuleWithProviders;
```
Do these changes in a separate commit

PhillipZada replied:
Hey Brandon, these changes are happening automatically as per the precommit script. Any suggestions on how to stop it.

brandonroberts replied:
Don't worry about it then. Must have merged another change that didn't get formatted.

PhillipZada commented:
Hey @brandonroberts,

The original request was to make it possible to create an adaptor using this 

```
export const myEntityAdapter: EntityAdapter<MyEntity> = createEntityAdapter<MyEntity>();
```

I removed the optional options flag and replaced it with a default value if not provided. Hope this meets requirements.

brandonroberts commented:
Thanks!
